Implemented Logit-Normal Distribution

ks147 opened the pull request:
<!-- BEGIN RELEASE NOTES -->
* stats
  * Implemented Logit-Normal probability distribution
<!-- END RELEASE NOTES -->

ks147 committed:
Implemented Logit-Normal Distribution

ks147 committed:
Made changes to pass quality test

ks147 committed:
changes to test_args to pass optional dependencies test

abh2k commented:
@czgdp1807 please review this pr.

czgdp1807 commented on sympy/stats/crv_types.py:
```diff
@@ -2449,6 +2451,93 @@ def LogLogistic(name, alpha, beta):
+
+    @staticmethod
+    def check(mu, s):
+        _value_check(s**2 > 0, "Scale parameter s must be positive.")
```
```suggestion
        _value_check((s**2).is_real is not False and s**2 > 0, "Squared scale parameter s must be positive.")
        _value_check(mu.is_real is not False, "Location parameter must be real")
```

czgdp1807 commented on sympy/stats/crv_types.py:
```diff
@@ -2449,6 +2451,93 @@ def LogLogistic(name, alpha, beta):
+    def check(mu, s):
+        _value_check(s**2 > 0, "Scale parameter s must be positive.")
+
+    def logit(self, x):
```
Shouldn't it be private?
```suggestion
    def _logit(self, x):
```

ks147 replied:
Yeah I've made the required changes in the latest commit

czgdp1807 commented on sympy/stats/crv_types.py:
```diff
@@ -2449,6 +2451,93 @@ def LogLogistic(name, alpha, beta):
+        mu, s = self.mu, self.s
+        return (S.One/2)*(1 + erf((self.logit(x) - mu)/(sqrt(2*s**2))))
+
+    def _moment_generating_function(self, t):
+        print("No analytical solution")
```
```suggestion
```
If no analytical solution is available, do not define that method in the subclass.

czgdp1807 commented on sympy/stats/crv_types.py:
```diff
@@ -2449,6 +2451,93 @@ def LogLogistic(name, alpha, beta):
+
+class LogitNormalDistribution(SingleContinuousDistribution):
+    _argnames = ('mu', 's')
+    set = Interval(0, 1)
```
```suggestion
    set = Interval.open(0, 1)
```
I believe `logit` isn't defined for `x = 0` and `x = 1`.

ks147 committed:
changes in LogitNormalDistribution:
-set the interval to (0,1) instead of [0,1]
-made logit(x) a private method
-removed mgf
-improved check method

abh2k commented:
@czgdp1807 @ks147  any updates.

ks147 commented:
> @czgdp1807 @ks147 any updates.

Yeah I made the required changes some time ago and commited them

czgdp1807 commented:
LGTM. Will merge tomorrow if no objections raised.
